### PR TITLE
Feature/anvil apps

### PR DESCRIPTION
**Pull request by bh412**

### CODE REVIEW

File: src/app/components/content/AnvilApp.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+});
+
+interface AnvilAppProps {
+    doc: AnvilAppDTO;
+    user: LoggedInUser | null;
+    pageState: any;
+}
+
+const AnvilAppComponent = ({doc, user, pageState}: AnvilAppProps) => {
+    const baseURL = `https://anvil.works/apps/${doc.appId}/${doc.appAccessKey}/app?s=new${Math.random()}`;
```
Comment (mlt47) [inline src/app/components/content/AnvilApp.tsx]: The session will change on every render, is that OK?

File: src/app/components/content/AnvilApp.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    }
+
+    if (parentQuestion !== undefined) {
+        if (parentQuestion.id != null) {
+            appParams["problem_id"] = parentQuestion.id;
+        }
+        if (parentQuestion.type != null) {
+            appParams["problem_type"] = parentQuestion.type;
+        }
+        parentQuestion.bestAttempt && parentQuestion.bestAttempt.correct && (appParams["problem_previously_correct"] = parentQuestion.bestAttempt.correct.toString());
```
Comment (jsharkey13) [inline src/app/components/content/AnvilApp.tsx]: Given the previous two are full `if` statements, why is this one not also?

File: src/app/components/content/AnvilApp.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    }
+
+    let queryParams = Object.keys(appParams).map((key) => {
+        return encodeURIComponent(key) + '=' + encodeURIComponent(appParams[key])
+    }).join('&');
+
+    let iframeSrc = `${baseURL}#?${queryParams}`;
+
+    let onMessage = function(e: any) {
+        if (iframeRef.current && e.source !== (iframeRef.current as HTMLIFrameElement).contentWindow) { return; }
```
Comment (jsharkey13) [inline src/app/components/content/AnvilApp.tsx]: Can this if body be on a new line?

File: src/app/components/content/AnvilApp.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+        let data = e.data;
+        console.debug("Anvil app message:", data);
```
Comment (jsharkey13) [inline src/app/components/content/AnvilApp.tsx]: I think we don't want this any more. Perhaps left commented out, as a prompt for debugging; but we don't want it usually.

File: src/app/components/elements/Accordion.tsx [hunk 5]
```diff
@@ -81,7 +82,7 @@ const AccordionComponent = ({id, trustedTitle, index, children, location: {hash}
         }
     }
 
-    return <div className="accordion">
+    return<div className="accordion">
```
Comment (jsharkey13) [inline src/app/components/elements/Accordion.tsx]: Any reason?

**Pull request merged**